src/cbc_sdk/platform/admin_grants.py

```diff
@@ -0,0 +1,471 @@
+
+        def set_roles(self, roles_list):
+            """
+            Sets the list of roles associated with the new profile.
+
+            Args:
+                roles_list (list): A list of role URNs.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._roles = roles_list
+            return self
+
+        def add_role(self, role):
+            """
+            Adds a role identifier to the list of roles associated with the new profile.
+
+            Args:
+                role (str): URN of the role to add.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._roles.append(role)
+            return self
+
+        def set_conditions(self, conditions_structure):
+            """
+            Sets the access conditions associated with the new profile.
+
+            Args:
+                conditions_structure (dict): The conditions associated with the new profile, with 'cidr', 'expiration',
+                                             and 'disabled' members.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions = conditions_structure
+            return self
+
+        def set_expiration(self, expiration):
+            """
+            Sets the expiration time on the new profile.
+
+            Args:
+                expiration (str): The expiration time, specified as ISO 8601.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['expiration'] = expiration
+            return self
+
+        def set_disabled(self, flag):
+            """
+            Sets whether or not the new profile is disabled.
+
+            Args:
+                flag (bool): True if this profile is disabled, False if noe.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['disabled'] = flag
+            return self
+
+        def set_can_manage(self, flag):
+            """
+            Sets the management flag for the new profile.
+
+            Args:
+                flag (bool): New value of the management flag.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._can_manage = flag
+            return self
+
+        def build(self):
+            """
+            Builds the new Profile object from the entered data.
+
+            Returns:
+                Profile: The new Profile object.
+            """
+            data = {'orgs': self._orgs, 'roles': self._roles, 'conditions': self._conditions,
+                    'can_manage': self._can_manage}
+            profile = Grant.Profile(self._grant._cb, self._grant, None, data)
+            self._grant._profiles.append(profile)
+            self._grant.touch()
+            return profile
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
```

Comment:
From an object structure do you mind putting the `__init__` function before the sub class Profile

Reply:
I think that actually _muddles_ the object structure to have `__init__` away from the rest of the `Profile` code.  But we'll do it your way.

Reply:
I know this is personal preference but I prefer being able to see how the variables are initialized next to the definition

Reply:
Already done.